map2loop/thickness_calculator.py

```diff
@@ -340,3 +347,208 @@ def compute(
+
+            GEO_SUB = geology[geology['UNITNAME'] == litho_in]['geometry'].values[0]
+            neighbour_list = list(
+                basal_contacts[GEO_SUB.intersects(basal_contacts.geometry)]['basal_unit']
+            )
+            B = calculate_endpoints(measurement_pt, strike, 10000, bbox_poly)
+            b = geopandas.GeoDataFrame({'geometry': [B]}).set_crs(basal_contacts.crs)
+
+            all_intersections = sampled_basal_contacts.overlay(
+                b, how='intersection', keep_geom_type=False
+            )
+            all_intersections = all_intersections[
+                all_intersections['geometry'].geom_type == 'Point'
+            ]
+
+            final_intersections = all_intersections[
+                all_intersections['basal_unit'].isin(neighbour_list)
+            ]
+
+            if 'MultiPoint' in final_intersections['geometry'].geom_type.values:
+                multi = final_intersections[
+                    final_intersections['geometry'].geom_type == 'MultiPoint'
+                ].index
+                for m in multi:
+                    nearest_ = shapely.ops.nearest_points(
+                        final_intersections.loc[m, :].geometry, measurement_pt
+                    )[0]
+                    final_intersections.at[m, 'geometry'] = nearest_
+                    final_intersections.at[m, 'geometry'] = nearest_
+
+            if len(final_intersections) < 2:
+                # print("Measurement n. ", s, "in litho ", litho_in, " has not enough intersections")
+                continue
+
+            if len(final_intersections['basal_unit'].unique()) == 1:
+                # print("Measurement n. ", s, "in litho ", litho_in, "does not cross two lithologies")
+                continue
+
+            int_pt1 = final_intersections.iloc[0].geometry
+            int_pt2 = final_intersections.iloc[1].geometry
+
+            if (
+                math.sqrt(((int_pt1.x - int_pt2.x) ** 2) + ((int_pt1.y - int_pt2.y) ** 2))
+                > map_dx / 2
+                or math.sqrt(((int_pt1.x - int_pt2.x) ** 2) + ((int_pt1.y - int_pt2.y) ** 2))
+                > map_dy / 2
+            ):
+                continue
+
+            seg1 = sampled_basal_contacts[
+                sampled_basal_contacts['basal_unit'] == final_intersections.iloc[0]['basal_unit']
+            ].geometry.iloc[0]
+            seg2 = sampled_basal_contacts[
+                sampled_basal_contacts['basal_unit'] == final_intersections.iloc[1]['basal_unit']
+            ].geometry.iloc[0]
+
+            if seg1.geom_type == 'MultiLineString':
+                seg1 = multiline_to_line(seg1)
+            if seg2.geom_type == 'MultiLineString':
+                seg2 = multiline_to_line(seg2)
+
+            strike1 = find_segment_strike_from_pt(seg1, int_pt1, measurement)
+            strike2 = find_segment_strike_from_pt(seg2, int_pt2, measurement)
+
+            b_s = strike - 30, strike + 30
```

Comment:
threshold parameter should be an attribute of the calculator and not hard coded

Reply:
in https://github.com/Loop3D/map2loop/pull/59/commits/80e4445bcfbe73f3a430c77dce0a96de56c913c6

Reply:
in https://github.com/Loop3D/map2loop/pull/59/commits/80e4445bcfbe73f3a430c77dce0a96de56c913c6